src/api/services/ingestion.go

```diff
@@ -669,13 +669,41 @@ func (i *IngestionService) ProcessVcf(
 							}
 						}
 
-						// TODO: improve 'alt' accuracy
 						//   By this point, tmpVariant["alt"] is populated with
 						//   an array of strings, i.e ["C", "CTT", "CTTTT", ...] .
 						//   Using the values in 'alleleLeft' and 'alleleRight' as
-						//   reference to which alleles are "most-likely", we
-						//   should remove (or store elsewhere) the values
-						//   corresponding to the "least-likely" alleles
+						//   reference to which alleles are "most-likely", remove
+						//   those that are "least-likely" here;
+
+						// indexing ref/alt in a vcf row:
+						//
+						// 		 0       1, 2, 3, ...
+						// ...  REF		ALT			...
+						// ...  G		CT,CTT,CTTT
+						if len(tmpVariant["alt"].([]string)) > 1 {
+							// hold a temporary copy of the current state of 'alt'
+							tmpVariantAltCopy := tmpVariant["alt"].([]string)
+
+							// built out a more optimal version of 'alt'
+							newVariantAlt := []string{}
+							// when 'alleleLeft' and '-Right' are both 0, set the
+							// 'alt' list to one element of '<NON_REF>'
+							// otherwise, preserve the allele(s) at position (x - 1)
+							// in the original 'alt' list,
+							if alleleLeft == 0 && alleleRight == 0 {
+								newVariantAlt = append(newVariantAlt, "<NON_REF>")
+							} else {
+								if alleleLeft > 0 {
+									newVariantAlt = append(newVariantAlt, tmpVariantAltCopy[alleleLeft-1])
+								}
+								if alleleRight > 0 {
+									newVariantAlt = append(newVariantAlt, tmpVariantAltCopy[alleleRight-1])
+								}
+							}
+
+							tmpVariant["alt"] = utils.RemoveDuplicates(newVariantAlt)
+						}
+						// else, don't touch 'tmpVariant["alt"]'
 
 						variation.Genotype = indexes.Genotype{
```

Comment:
rather than modifying the row in the VCF, `GT` on each sample should set the actual alleles on the variation for this particular sample - I may be misunderstanding the Go here though.

for a concrete example:

if a row has something like `chr1:1005 REF=A, ALTS=C,CC,CA SAMPLE1:GT=0/1`

then the genotype for that individual is heterozygous, and the alleles for that sample at that variant are `[A, C]`. If the user searches for a `C` allele, they don't mean there is a `C` in the alt field in the VCF variant line. Rather, they are interested in samples with a `C` allele in the allele array, at the location specified by the variant - in this case, they are interested in a sample with `1` in the `GT` field.  

even if the other alts on this variant are never used, i'm not sure they should need to be touched, since they're referred to by other fields (probabilities & what have you which are extraneous but we can preserve them anyway) - so I don't think they need to be stripped out, since they shouldn't even be referred to except during the initial build of the genotype / probability info for each sample.

Reply:
in terms of the models in the indexes file in Gohan, this would probably mean putting the actual alleles array into `Sample.Variation.Genotype` somewhere - since the alleles in question `[A, C]` make up the genotype of the sample.